distributor/src/distributor/messenger.bal

```diff
@@ -43,96 +42,99 @@ function getAwaitResultsMessage(string electionCode, string resultType, string r
+        }
+    }
+}
+
+# Validate and sanitize local mobile number into the proper format.(0771234567).
+#
+# + mobileNo - User provided mobile number
+# + return - Formatted mobile number or the error
+function validate(string mobileNo) returns string|error {
+    string mobile = <@untained> mobileNo.trim();
+
+    if (mobile.startsWith("+94") && mobile.length() == 12) {
+        mobile = mobile.substring(1);
+    }
+
+    boolean number = stringutils:matches(mobile, "^[0-9]*$");
+    if !number {
+        return error(ERROR_REASON, message = "Invalid mobile number. Given mobile number contains non numeric " +
+                                           "characters: " + mobile);
+    }
+
+    if (mobile.startsWith("0") && mobile.length() == 10) {
+        return mobile;
+    }
+    if (mobile.startsWith("94") && mobile.length() == 11) {
+        return mobile;
+    }
+    // Allow only the local mobile numbers to register via public API. International number are avoided.
+    return error(ERROR_REASON, message = "Invalid mobile number. Resend the request as follows: If the " +
+                                     "mobile no is 0771234567, send POST request to  \'/sms\' with JSON payload " +
+                                     "\'{\"username\":\"myuser\", \"mobile\":\"0771234567\"}\'");
+}
+
+# Register recipient in the mobileSubscribers list and persist in the smsRecipients db table.
+#
+# + username - The recipient username
+# + mobileNo - The recipient number
+# + return - The status of registration or operation error
+function registerAsSMSRecipient(string username, string mobileNo) returns string|error {
+
+    if mobileSubscribers.hasKey(username) {
+        string errMsg = "Registration failed: username:" + username + " is already registered with mobile:" + mobileNo;
+        log:printError(errMsg);
+        return error(ERROR_REASON, message = errMsg);
+    }
+
+    // Persist recipient number in database
+    var status = dbClient->update(INSERT_RECIPIENT, username, mobileNo);
+    if status is error {
+        log:printError("Failed to persist recipient no in database", status);
+        return error(ERROR_REASON, message = "Registration failed: username:" + username + " mobile:" + mobileNo
+                                         + ": " + <string> status.detail()?.message);
+    }
+    mobileSubscribers[username] = mobileNo;
+
+    return "Successfully registered: username:" + username + " mobile:"  + mobileNo;
+}
+
+# Unregister recipient from the mobileSubscribers map and remove from the smsRecipients db table.
+#
+# + username - The recipient username
+# + mobileNo - The recipient number
+# + return - The status of deregistration or operation error
+function unregisterAsSMSRecipient(string username, string mobileNo) returns string|error {
+
+    // Remove persisted recipient number from database
+    var status = dbClient->update(DELETE_RECIPIENT, username, mobileNo);
```

Comment:
If one user can have only one associated mobile number, shall we first check if the in-memory map matches the number?

If it doesn't, that's an error and we return there.

Then we can remove from the db by passing only the username right?